app/forms/work_deposit_pathway.rb

```diff
@@ -0,0 +1,151 @@
+# frozen_string_literal: true
+
+class WorkDepositPathway
+  def initialize(resource)
+    @resource = resource
+  end
+
+  def details_form
+    if scholarly_works?
+      ScholarlyWorks::DetailsForm.new(resource)
+    elsif general?
+      General::DetailsForm.new(resource)
+    else
+      resource
+    end
+  end
+
+  def publish_form
+    if scholarly_works?
+      ScholarlyWorks::PublishForm.new(resource)
+    else
+      resource
+    end
+  end
+
+  private
+
+    attr_reader :resource
+
+    delegate :work_type, to: :resource
+
+    # It will probably make sense to promote these predicate methods to the
+    # public interface at some point.
+    def scholarly_works?
+      Work::Types.scholarly_works.include?(work_type)
+    end
+
+    def general?
+      Work::Types.general.include?(work_type)
+    end
+
+    class DetailsFormBase
+      include ActiveModel::Model
+      include ActiveModel::Attributes
+
+      COMMON_FIELDS = %w{
+        description
+        published_date
+        subtitle
+        publisher_statement
+        keyword
+        publisher
+        identifier
+        related_url
+        subject
+        language
+      }.freeze
+
+      def initialize(work_version)
+        @work_version = work_version
+        super(work_version.attributes.slice(*self.class.form_fields))
+      end
+
+      validates :description, presence: true
+
+      validates :published_date,
+                presence: true,
+                edtf_date: true
+
+      def self.model_name
+        WorkVersion.model_name
+      end
+
+      def save(context:)
+        if valid?
+          work_version.attributes = attributes
+          work_version.save(context: context)
+        end
+      end
+
+      delegate :id,
+               :to_param,
+               :persisted?,
+               :uuid,
+               :new_record?,
+               :published?,
+               :draft?,
+               :work,
+               :indexing_source=,
+               :update_doi=,
+               to: :work_version, prefix: false
+
+      private
+
+        attr_reader :work_version
+    end
+
+    module General
+      class DetailsForm < DetailsFormBase
+        def self.form_fields
+          DetailsFormBase::COMMON_FIELDS.union(
+            %w{
+              based_near
+              source
+              version_name
+            }
+          ).freeze
+        end
+
+        form_fields.each { |attr_name| attribute attr_name }
+
+        validates :version_name,
+                  allow_blank: true,
+                  format: {
+                    with: /^(0|[1-9]\d*)\.(0|[1-9]\d*)\.(0|[1-9]\d*)(?:-((?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+([0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?$/, # rubocop:disable Layout/LineLength
+                    message: I18n.t('activerecord.errors.work_version.attributes.version_name.format'),
```

Comment:
The path to this translation is missing `.models`: `I18n.t('activerecord.errors.models.work_version.attributes.version_name.format')`

Also, I'm curious why this validation is here.  It's basically the same as the model level validation except for `allow_blank: true`.

Reply:
@ajkiessl Thanks for not just reviewing the code but also doing thorough QA - very much appreciated!

You're right. I just needed the form to reflect the underlying model validation here since the rules are identical, but there are better ways to do that than what I did (duplicating the same validation logic).